### PR TITLE
Integration Tests - Actions / Docker / Terraform / TestIM

### PR DESCRIPTION
**Pull request by BrandonSharratt**
First large pass at integration tests.
They add their own github action workflow
Terraform is used to stand up docker containers for the backend, frontend and postgres
Secrets configure the authentication/user

Integration tests run in testim (https://www.testim.io/) they are therefore configured there, and can be updated independently. Screenshots of failures/successes and test logs are also available over there.

The following repository VARIABLES are required
- TESTIM_PROJECT - this is the project in testim (you can see this in the forked repository)
- TESTIM_USER - this is the user in testim (you can also see this in the forked repository)

The following repository SECRETS are required
 - TESTIM_TOKEN - the testim secret token required to run the tests via the CLI
 - OIDC_URL - the issuer url
 - OIDC_CLIENT - the client that the api uses for oidc
 - OIDC_CLIENT_SECRET - the client secret the api uses for oidc
 - AUTH_API_URL - the full url including version of the auth api
 - PAY_API_URL - the full url including version of the pay api
 - TEST_USER - the user credentials to login to the application using username/password
 - TEST_PASSWORD - the user credentials to login to the application using username/password
 - CANADA_POST_API_KEY - the api key the frontend uses for canada post api
 - AUTH_URL - the auth url ending with /auth ie http://keycloak.com/auth
 - AUTH_REALM - the realm for the above url
 - REGISTRY_HOME_URL - the registry home url from the frontend configuration
 - NUXT_CLIENT_ID - the client id that the frontend uses for oidc

### CODE REVIEW

**Comment by BrandonSharratt** [comment]
As far as I can tell the checks are failing due to code that is not changed in this PR

**Comment by bcregistry-sre** [comment]
How do we know the test is success or failure?

File: .github/workflows/integration-test.yaml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+          docker build . -t regweb:latest
+          cd $WD/terraform
+          mkdir _tmp
+          chmod -R 777 _tmp
+          
+          echo "127.0.0.1 ar.example.demo" | sudo tee -a /etc/hosts
+          echo "127.0.0.1 ar.example.demo" | sudo tee -a /etc/hosts
+          terraform init
+
+          #create terraform vars file
```
Comment (bcregistry-sre) [inline .github/workflows/integration-test.yaml:64]: For this part consider using the 1password Vault service or google secret manager to retrieve the value. And don't need to put so many values in the github.

**Comment by BrandonSharratt** [comment]
> How do we know the test is success or failure?

If the test is a failure the entire job is a failure and then you can either reference this workflow as a requirement or add to this workflow so that you don't deploy broken builds

**Pull request merged**